migrations_postgres/2023-09-07-131701_create_db/up.sql

```diff
@@ -0,0 +1,22 @@
+-- Your SQL goes here
+
+CREATE TABLE ownership_voucher (
+    guid varchar(36) NOT NULL PRIMARY KEY,
+    contents bytea NOT NULL
+);
+
+CREATE TABLE manufacturer_vouchers (
+    ov_guid varchar(36) PRIMARY KEY references ownership_voucher(guid) ON DELETE CASCADE
+);
+
+CREATE TABLE owner_vouchers (
+    ov_guid varchar(36) PRIMARY KEY references ownership_voucher(guid) ON DELETE CASCADE,
+    to2_performed boolean,
+    to0_accept_owner_wait_seconds bigint
+);
+
+CREATE TABLE rendezvous_vouchers (
+    ov_guid varchar(36) PRIMARY KEY references ownership_voucher(guid) ON DELETE CASCADE,
+    ttl bigint
+);
+
```

Comment:
what if instead of 3 more tables, we have attributes only in the ownership_voucher table? the fact that to2 has been performed is a field of the OV itlself, same for the owner_wait_seconds and the ttl once it's sent to the rendezvous - I have not yet thought how this would look like but just something to think about (I came up with this questions because it seems a repetition when we could instead play with attributes on the OV itself as that's what we really deal with)

Reply:
I went with that option first, but if we separate things by table we are going to be able to give separate permissions and allow each server to access just what it needs, there are fields that shouldn't be touched by other servers.
We can handle those permissions if we have separate tables, but (to the best of my knowledge) we cannot do that easily if we have a single table.

Also, those fields are not OV fields "inside the OV", but extended attributes that we add to the OV file. Those extended attributes only appear on the OV when it is stored in a specific server, aka ttl is just on the rendezvous server and so on.

Reply:
and at this point, if the above is true, then we'd need to have a schema for each server, at which point, it might as well just be called ownership_voucher for all the servers - what bothers me is that we are defining just one schema with tables named differently for things that won't possibly even run on the same db.
For instance, I want to act only as rendezvous server, I don't need the "extra" tables and I only want to write OV+ttl - I might as well just have a table named ownership_voucher anyway but we're bundling everything together like this :thinking:

Reply:
yep. I was making the assumption of a single database server running somewhere were all the servers could connect to. But that is not something that works if we want everything to be independent. So, we need three different database initialization scripts since there are going to be as many databases as server types (maybe owner+service info api are a single entity). And there, yes a single table with the fields that each server needs.

Reply:
thanks for the comments, I was making a wrong assumption :pray: 
Will update accordingly

Reply:
cool cool - I actually raised this up for discussion - my assumption is that servers can run completely indipendenly from each other (surely for manufacturer and the rest) - we can discuss it further if we think that's not the case